### ISSUE TITLE
ATLAS PESSTO outgest issues

### ISSUE THREAD

**Issue by genghisken** [genghisken/ps1#55]
Originally reported by: **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**

----------------------------------------

I rewrote the PESSTO outgest code to be rendered by Django internally.  So all PESSTO has to do is hit the following links:

http://psweb.mp.qub.ac.uk/sne/atlas3/pesstorecurrences/

http://psweb.mp.qub.ac.uk/sne/atlas3/pesstosummary/

The problem is that Django takes an enormous length of time to render these simple plain text rows.  I use Django-Tables2 to render the rows.  The database query takes < 0.5 seconds, so something is happening with access to the internal Django table to slow it down to ~ 60 seconds (for 8500 rows).

Options are:

* Bypass Django-Tables2 and render the rows direct from the queryset.  (A bit more work to order the data.)
* Use a separate script to render the summary.txt and recurrences.txt, which will NOT be dynamic.
* Find out what inside Django-Tables2 is slowing down the rendering.

----------------------------------------
- Bitbucket: https://bitbucket.org/genghisken/ps1/issue/55

**Comment by genghisken** [comment]
*Original comment by* **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**:

----------------------------------------

After another test, it looks like this is a fundamental limitation in Django-Tables(2), since attempting to render 7000 ordinary HTML rows also slows it down enormously.

**Comment by genghisken** [comment]
*Original comment by* **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**:

----------------------------------------

A switch to bypassing Django-Tables2 and using the queryset directly did speed things up (at least double the speed).  However, when given a set of 40,000 objects, we had the following issues:

* gateway timeout
* oversized headers error

In other words, Apache / WSGI just can't handle the lightcurve points associated with 40,000 objects.  We (alas) will eventually have to resort to the old mechanism of generating csv files by script.

**Comment by genghisken** [comment]
We now have database workarounds, but we need to stop using Django templates to generate large text files. It's just too slow for that, and takes far too much memory on the webserver.
